app/uk/gov/hmrc/nisp/controllers/auth/AuthorisedForNisp.scala

```diff
@@ -65,16 +68,13 @@ trait AuthorisedForNisp extends Actions {
   object AuthorisedByAny extends AuthorisedBy(NispAnyRegime)
   object AuthorisedByVerify extends AuthorisedBy(NispVerifyRegime)
 
-  def retrieveName(authContext: AuthContext)(implicit request: Request[AnyContent]): Future[Option[String]] = {
+  def retrievePerson(authContext: AuthContext)(implicit request: Request[AnyContent]): Future[Option[Citizen]] = {
     val nino = retrieveNino(authContext.principal)
     citizenDetailsService.retrievePerson(nino).map { citizenOption =>
       for (
-        citizen <- citizenOption;
-        name <- citizen.name;
-        firstName <- name.firstName;
-        lastName <- name.lastName
+        citizen <- citizenOption
       ) yield {
-        s"$firstName $lastName"
+        citizen
```

Comment:
I don't think this for comprehension is still necessary. The method could probably be refactored into one line.